lib/cloudware/azure.rb

```diff
@@ -142,14 +143,30 @@ def machines
                                provider: 'azure',
                                type: r.tags['cloudware_machine_type'],
                                flavour: r.tags['cloudware_machine_flavour'],
-                               state: get_instance_state(r.tags['cloudware_domain'], r.tags['cloudware_machine_name'])
+                               state: machine_state(r.tags['cloudware_machine_name'], r.tags['cloudware_domain'])
                              })
           end
         end
         @machines
       end
     end
 
+    def machine_state(name, domain)
+      state = @compute_client.virtual_machines.instance_view("#{domain}-#{name}", name)
+      return 'running' unless state.statuses.find { |s| s.code =~ /PowerState\/running/ }.nil?
+      return 'stopped' unless state.statuses.find { |s| s.code =~ /PowerState\/stopped/ }.nil?
```

Comment:
I'd extract a method for the duplicated `state.statuses.find { |s| s.code =~ /PowerState\/running/ }.nil?` check.